processor/maskprocessor/README.md

```diff
@@ -0,0 +1,79 @@
+# Mask Processor
+This processor is used to detect and mask sensitive data.
+
+## Supported pipelines
+- Logs
+- Metrics
+- Traces
+
+## How It Works
+1. This processor traverses the attribute and resource fields of incoming telemetry. For log-based telemetry, it will also traverse the body.
+2. If a field matches a defined regex rule, the matching value is replaced with `[masked_value]`, where `value` is the name of the  rule. For instance, a rule that masks email addresses would result in `[masked_email]`.
+3. If a field is to be excluded from masking, it can be specified in the processor's `exclude` field. By default, the processor will mask all fields.
+
+## Configuration
+| Field       | Default | Description |
```

Comment:
One of the new documentation standards items we've decided on is to include data type for these fields. It is the only thing "missing" from this doc based on the new standards Craig and I are writing. I realize that to some this is redundant, but not everyone is capable of inferring allowed data types from the examples.

Reply:
@Dylan-M Applied types to the readme.